Pod/Classes/NJHNavigationBarSelectorPageViewController.m

```diff
@@ -0,0 +1,228 @@
+//
+//  NJHNavigationBarSelectorPageViewController.m
+//  Pods
+//
+//  Created by Hakon Hanesand on 7/29/15.
+//
+
+#import "NJHNavigationBarSelectorPageViewController.h"
+#import "NJHNavigationBarBackgroundSelectionView.h"
+
+/**
+ *  The ratio of the background selection view's height to its super view height
+ */
+static double const kNJHBackgroundSelectionViewVerticalRatio = 0.7;
+
+/**
+ *  The ratio of the background selection view's width to its super view width
+ */
+static double const kNJHBackgroundSelectionViewHorizontalRatio = 0.65;
+
+@interface NJHNavigationBarSelectorPageViewController ()
+@property (nonatomic) UINavigationItem *targetNavigationItem;
+@property (nonatomic) NSMutableArray *viewControllerArray;
+@property (nonatomic, getter=isPageScrolling) BOOL pageScrolling;
+@property (nonatomic, getter=hasInitialized) BOOL initialized;
+@property (nonatomic, readwrite) NSInteger currentPageIndex;
+@end
+
+@implementation NJHNavigationBarSelectorPageViewController
+
+- (instancetype)initWithPageViewControllers:(NSArray *)pageViewControllers navigationItem:(UINavigationItem *)navigationItem {
+    if (self = [super initWithTransitionStyle:UIPageViewControllerTransitionStyleScroll navigationOrientation:UIPageViewControllerNavigationOrientationHorizontal options:nil]) {
+        if (!navigationItem) {
+            self.targetNavigationItem = self.navigationItem;
+        } else {
+            self.targetNavigationItem = navigationItem;
+        }
+        
+        [self.viewControllerArray addObjectsFromArray:pageViewControllers];
+        self.delegate = self;
+        self.dataSource = self;
```

Comment:
Not important since you have done a good job at keeping this VC small, but I would normally separate away this responsibility from the view controller.